sktime/lagged_model_validator.py

```diff
@@ -0,0 +1,276 @@
+            self.test_estimator.lagtime = lag
+            self.test_estimator.fit(data)
+            estimated_models.append(self.test_estimator.fetch_model().copy())
+
+        for mlag in self.mlags:
+            if mlag == 0: continue
+            # make a prediction using the test model
+            predictions.append(self._compute_observables(self.test_model, mlag=mlag))
+            # compute prediction errors if we can
+            if self.has_errors:
+                l, r = self._compute_observables_conf(self.test_model, mlag=mlag)
+                predictions_conf.append((l, r))
+
+        for model in estimated_models:
+            if model is None: continue
+            # evaluate the estimate at lagtime*mlag
+            estimates.append(self._compute_observables(model))
+            if self.has_errors and self.err_est:
+                l, r = self._compute_observables_conf(model)
+                estimates_conf.append((l, r))
+
+        # build arrays
+        estimates = np.array(estimates)
+        predictions = np.array(predictions)
+        if self.has_errors:
+            predictions_conf = np.array(predictions_conf)
+        else:
+            predictions_conf = np.array((None, None))
+        if self.has_errors and self.err_est:
+            estimates_conf = np.array(estimates_conf)
+        else:
+            estimates_conf = np.array((None, None))
+
+        m = self._model
+        m._estimates = estimates
+        m._estimates_conf = estimates_conf
+        m._predictions = predictions
+        m._predictions_conf = predictions_conf
+        m._lagtimes = lags
+
+        return self
+
+    @abc.abstractmethod
+    def _compute_observables(self, model, mlag=1):
+        """Compute observables for given model
+
+        Parameters
+        ----------
+        model : Model
+            model to compute observable for.
+
+        mlag : int, default=1
+            if 1, just compute the observable for given model. If not 1, use
+            model to predict result at multiple of given model lagtime.
+
+        Returns
+        -------
+        Y : ndarray
+            array with results
+
+        """
+        pass
+
+    @abc.abstractmethod
+    def _compute_observables_conf(self, model, mlag=1, conf=0.95):
+        """Compute confidence interval for observables for given model
+
+        Parameters
+        ----------
+        model : Model
+            model to compute observable for. model can be None if mlag=0.
+            This scenario must be handled.
+
+        mlag : int, default=1
+            if 1, just compute the observable for given model. If not 1, use
+            model to predict result at multiple of given model lagtime.
+
+        Returns
+        -------
+        L : ndarray
+            array with lower confidence bounds
+        R : ndarray
+            array with upper confidence bounds
+
+        """
+        pass
+
+    @abc.abstractmethod
+    def _observable_dummy_mlag0(self, conf=False):
```

Comment:
do we still need this?

Reply:
I thought it might be a good idea to get rid of this computation in the main worker (the need to implement the case for mlag=0 was just mentioned in the docstring), but this pattern forces the implementer of the class to handle it explicitly.

Reply:
We could probably clean the code up a lot, if we would just raise.

Reply:
i am in favor of raising, lagtime is only well defined for values > 0.

Reply:
done. But one test in msm cktest fails with a precision violation, which I can not spot right now. I'd be delighted, if you could take a look!

Reply:
Sure, I will have a look